From cc988eb9790faf8a48b4ce636615beed535da282 Mon Sep 17 00:00:00 2001
From: Meng Li <Meng.Li@windriver.com>
Date: Thu, 14 Dec 2023 16:08:43 +0800
Subject: [PATCH] s32g: hse: improve code of secure boot enable command to
 compatible with BSP39

In order to enable hse secure boot feature from u-boot, the related code is
reverted from u-boot 2022.04. But when this BSP nxp-s32g is upgraded to NXP
BSP39, the u-boot version is 2022.04, so there are some errors when building
hse secure boot related code. Therefore, create a patch to fix these building
errors.

Upstream-Status: Inappropriate [WR Linux specific]

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-s32/s32-cc/hse_adv_secboot.c | 8 +++++---
 include/hse/hse_abi.h                      | 1 +
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-s32/s32-cc/hse_adv_secboot.c b/arch/arm/mach-s32/s32-cc/hse_adv_secboot.c
index 32e8e3d95e..6e4f4d0512 100644
--- a/arch/arm/mach-s32/s32-cc/hse_adv_secboot.c
+++ b/arch/arm/mach-s32/s32-cc/hse_adv_secboot.c
@@ -13,8 +13,10 @@
 #include <malloc.h>
 #include <hse/hse_abi.h>
 #include <hse/hse_mu.h>
+#include <fdt_support.h>
+#include <asm/global_data.h>
 
-#ifdef CONFIG_SD_BOOT
+#if defined(CONFIG_SD_BOOT) || defined(CONFIG_QSPI_BOOT)
 DECLARE_GLOBAL_DATA_PTR;
 
 /* hse nvm key catalog configuration */
@@ -967,7 +969,7 @@ static int get_public_key(struct hse_private *priv, char *pubkey_file)
 }
 
 
-static int do_hse_secboot_enable(cmd_tbl_t *cmdtp, int flag,
+static int do_hse_secboot_enable(struct cmd_tbl *cmdtp, int flag,
 				 int argc, char * const argv[])
 {
 	struct hse_private *priv;
@@ -1079,7 +1081,7 @@ U_BOOT_CMD(hse_secboot_enable, 2, 0, do_hse_secboot_enable,
 	   "generate device-specific SYS_IMG",
 	   "Generate SYS-IMG and place it on SD card");
 
-static int do_hse_keystore_format(cmd_tbl_t *cmdtp, int flag,
+static int do_hse_keystore_format(struct cmd_tbl *cmdtp, int flag,
 				  int argc, char * const argv[])
 {
 	struct hse_private *priv;
diff --git a/include/hse/hse_abi.h b/include/hse/hse_abi.h
index c8a308e5a9..aa5116b31a 100644
--- a/include/hse/hse_abi.h
+++ b/include/hse/hse_abi.h
@@ -11,6 +11,7 @@
 #include <common.h>
 #include <mmc.h>
 #include <uuid.h>
+#include <command.h>
 
 #define BYTES_TO_BITS(x)	((x) * 8u)
 
-- 
2.34.1

